<launch>
  <arg name="config_file" default="$(find-pkg-share pose_initializer)/config/pose_initializer.param.yaml"/>

  <!--
    <remap from="service/initialize_pose" to="/localization/util/initialize_pose"/>
    <remap from="service/initialize_pose_auto" to="/localization/util/initialize_pose_auto"/>

    <param name="use_gnss" value="true"/>
    <remap from="gnss_pose_cov" to="/sensing/gnss/pose_with_covariance"/>
    <remap from="pointcloud_map" to="/map/pointcloud_map"

    <param from="$(var config_file)"/>
    <param name="enable_gnss_callback" value="true"/>
  -->

  <node pkg="pose_initializer" exec="pose_initializer_node" name="pose_initializer_node" output="log">
    <param from="$(var config_file)"/>
    <remap from="ndt_align_srv" to="/localization/pose_estimator/ndt_align_srv"/>
    <remap from="ekf_reset_srv" to="/initialpose3d"/>
  </node>

  <node pkg="pose_initializer" exec="pose_initializer_rviz" name="pose_initializer_rviz" output="log">
    <remap from="initialpose" to="/initialpose"/>
    <remap from="pointcloud_map" to="/map/pointcloud_map"/>
  </node>

  <!--
  <node pkg="pose_initializer" exec="pose_initializer_rviz" name="pose_initializer_auto" output="log">
  </node>
  -->
</launch>
